dialogue.py

```diff
@@ -0,0 +1,283 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Serializes/deserializes json objects containing run-type configurations.
+# Invokes run.sh with configurations stored in the file
+# 'runtype_options.json'
+
+
+import json
+
+def write_json(filename, python_dict):
+    """
+    Serialize python_dict (dictionary) to filename (text file).
+    :param filename: string
+    :param python_dict: dict
+    """
+    with open(filename, 'w') as f:
+        json.dump(python_dict, f, indent=4)
+
+def read_json(filename):
+    """
+    Deserializes json formatted string from filename (text file) as a dictionary.
+    :param filename: string
+    """
+    with open(filename) as f:
+        return json.load(f)
+
+def create_run_dict(runtype, *args):
+    """
+    Returns a dictionary of a run-type option object deserialized
+    from a human-readable configuration file (e.g., JSON, HJSON, YAML).
+    :param runtype: (HBIR || HBIR_RT || PRESET || LOADLEVEL)
+    """
+    runtype_option_dict = read_json('runtype_options.json')
+    if runtype not in runtype_option_dict:
+        return {}
+    parameter_list = runtype_option_dict[runtype]
+
+    if(len(parameter_list) == len(args)):
+        return_dict = {tup[0] : tup[1] for tup in zip(parameter_list, args)}
+        return_dict['Run Type'] = runtype
+        return return_dict
+    return {}
+
+def create_run_dict_list(runtype, args):
+    """
+    Returns a dictionary of run-type option objects.
+    :param runtype: (HBIR || HBIR_RT || PRESET || LOADLEVEL)
+    """
+    return create_run_dict(runtype, *args)
+
+def create_run_sh_invocation(run_dict):
+    """
+    Builds a string to be used to invoke the `run.sh` script in ~/workloads. The
+    command-line arguments of `run.sh` are initialized with the respective values in
+    run_dict.
+    :param run_dict: dict
+    """
+    try:
+        runtype_option_dict = read_json('runtype_options.json')
+        parameters = runtype_option_dict[run_dict['Run Type']]
+        return './run.sh {}'.format(
+            ' '.join(str(run_dict[p]) for p in parameters))
+    except: # Probably a dictionary failure.
+        return ''
+
+
+def execute_run(run_dict):
+    """
+    Invoke the `run.sh` script in ~/workloads with command-line arguments stored
+    in run_dict.
+    :param run_dict: dict
+    """
+    # We'll eventually invoke this for realsies. For now, we're just going to
```

Comment:
@mbottini we should add a PR at some point to join the GUI and the existing blackbox implementation.

Reply:
@Zonr0 Made a comment on moving the GUI to its own repo, going to have this be a TODO for now.